DependencyInjection/Compiler/WellKnownSchemaFilterPass.php

```diff
@@ -33,26 +34,32 @@ public function process(ContainerBuilder $container)
                 continue;
             }
 
+            $table = null;
+
             switch ($definition->getClass()) {
                 case DoctrineDbalAdapter::class:
                 case PdoAdapter::class:
-                    $blacklist[] = $definition->getArguments()[3]['db_table'] ?? 'cache_items';
+                    $table = $definition->getArguments()[3]['db_table'] ?? 'cache_items';
                     break;
 
                 case PdoSessionHandler::class:
-                    $blacklist[] = $definition->getArguments()[1]['db_table'] ?? 'sessions';
+                    $table = $definition->getArguments()[1]['db_table'] ?? 'sessions';
                     break;
 
                 case DoctrineDbalStore::class:
                 case PdoStore::class:
-                    $blacklist[] = $definition->getArguments()[1]['db_table'] ?? 'lock_keys';
+                    $table = $definition->getArguments()[1]['db_table'] ?? 'lock_keys';
                     break;
 
                 case LegacyConnection::class:
                 case Connection::class:
-                    $blacklist[] = $definition->getArguments()[0]['table_name'] ?? 'messenger_messages';
+                    $table = $definition->getArguments()[0]['table_name'] ?? 'messenger_messages';
                     break;
             }
+
+            if ($table && ! method_exists($definition->getClass(), 'configureSchema')) {
+                $blacklist[] = $table;
```

Comment:
This pass is completely broken. It checks for services' class, but many of the services above are not exposed by class-name:
- cache adapters are ChildDefinition at this stage, which means they don't have a class yet
- lock stores are created by a factory
- messenger connections are also created by a factory

Only sessions are filtered by this pass.
PdoSessionHandler::configureSchema is a new method provided by https://github.com/symfony/symfony/pull/48059

I'd be tempted to remove all the logic but this check for PdoSessionHandler, and adjust tests accordingly.

Would that be OK?

Reply:
Makes sense to me